errai-cdi/errai-cdi-server/src/test/java/org/jboss/errai/cdi/injection/client/test/InstanceInjectionIntegrationTest.java

```diff
@@ -133,4 +133,12 @@ public void testIsAmbiguous() {
     assertFalse(instanceDependentBeanA.isAmbiguous());
   }
 
+  public void testIterator() {
+    final DependentInstanceTestBean testBean
+            = getBeanManager().lookupBean(DependentInstanceTestBean.class).getInstance();
+
+    final Instance<DependentBeanA> injectedBean = testBean.getInjectDependentBeanA();
+
+    assertTrue(injectedBean.iterator().hasNext());
```

Comment:
Glad to see test coverage, but this test is missing some assertions that would make it much more robust. In particular, it should call `Iterator#next()` and verify that the returned value is a `DependentBeanA`. This would be sufficient to reproduce the issue I described above.